src/ddprof_module_lib.cc

```diff
@@ -6,12 +6,106 @@
 #include "ddprof_module_lib.hpp"
 
 #include "ddres.hpp"
+#include "defer.hpp"
+#include "failed_assumption.hpp"
 #include "logger.hpp"
+#include "string_format.hpp"
+
+#include <fcntl.h>
+#include <gelf.h>
+#include <libelf.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
 
 namespace ddprof {
 
-DDRes update_module(Dwfl *dwfl, ProcessAddress_t pc,
-                    const DDProfModRange &mod_range,
+static bool get_elf_offsets(const std::string &filepath, Offset_t &start_offset,
+                            Offset_t &bias_offset) {
+  int fd = ::open(filepath.c_str(), O_RDONLY);
```

Comment:
we could have a single fd per file that we keep open and give to dwfl (this seems to generate less file opens)
I'll add that in later commit